### PR TITLE
feat: introduce shared/http-request base class

### PR DESCRIPTION
**Pull request by eoln**

### CODE REVIEW

File: src/models/inbound/backend-requests.ts [hunk 1]
```diff
@@ -47,126 +38,25 @@ import {
...
-
-  logger: SDKLogger.Logger
-
-  // type of http scheme
-  scheme: Scheme
-
-  // target uri for all requests
-  uri: string
-
+export interface BackendConfig extends HttpRequestConfig{
```
Comment (lewisdaly) [inline src/models/inbound/backend-requests.ts:41]: nit: missing space before `{`

File: src/models/inbound/backend-requests.ts [hunk 2]
```diff
@@ -47,126 +38,25 @@ import {
...
-  async put<Body, Response> (path: string, body: Body): Promise<Response | void> {
-    return this.loggedRequest({
-      uri: this.fullUri(path),
-      method: 'PUT',
-      body: requests.common.bodyStringifier(body),
-      headers: this.headers,
-      agent: this.agent
-    })
+  // GETTERS
+  get singAuthorizationPath (): string {
```
Comment (lewisdaly) [inline src/models/inbound/backend-requests.ts:58]: nit: should this be `signAuthorizationPath`?

**Review by lewisdaly — CHANGES_REQUESTED** [review]
Looks good - minor typos to fix I think

File: src/models/inbound/backend-requests.ts (hunk 1 above)
Comment (eoln) [reply src/models/inbound/backend-requests.ts]: fixed

File: src/models/inbound/backend-requests.ts (hunk 2 above)
Comment (eoln) [reply src/models/inbound/backend-requests.ts]: fixed

**Review by lewisdaly — APPROVED** [review]

**Pull request merged**